### PR TITLE
Exclude tmpcharts folder and helm generated .lock files from list of watched files

### PR DESCRIPTION
**Pull request by gsquared94**
Exclude `tmpcharts` folder and helm generated `.lock` files from list of watched files

Fixes: #3565  
**Description**

`helm dep up` makes changes to the `tmpcharts` in addition to `charts` subdirectory within a chart package. It also modifies `requirements.lock` file (helm 2) and `Chart.lock` file (helm 3). So we exclude these file from the file change monitor when `skipBuildDependencies` is not set to `true`

This is consistent with existing behavior of excluding `charts` subdirectory similarly.

### CODE REVIEW

File: pkg/skaffold/deploy/helm.go [hunk 1]
```diff
@@ -162,14 +162,43 @@ func (h *HelmDeployer) Dependencies() ([]string, error) {
...
 		isDep := func(path string, info walk.Dirent) (bool, error) {
-			return !info.IsDir() && (!strings.HasPrefix(path, chartDepsDir) || r.SkipBuildDependencies), nil
+			if info.IsDir() {
+				return false, nil
+			}
+			if r.SkipBuildDependencies {
+				return true, nil
+			}
+
+			for _, v := range chartDepsDirs {
```
Comment (tejal29) [inline pkg/skaffold/deploy/helm.go:189]: Would this miss any other `requirements.lock` say in a python app?

File: pkg/skaffold/deploy/helm.go [hunk 2]
```diff
@@ -162,14 +162,43 @@ func (h *HelmDeployer) Dependencies() ([]string, error) {
...
+				if strings.HasPrefix(path, filepath.Join(release.ChartPath, v)) {
+					return false, nil
+				}
+			}
+
+			for _, v := range lockFiles {
+				if strings.EqualFold(info.Name(), v) {
```
Comment (tejal29) [inline pkg/skaffold/deploy/helm.go:196]: Would this miss any other requirements.lock say in a python app? 
Shd we look for `reuirements.lock` in the chartPath?

Comment (tejal29) [reply pkg/skaffold/deploy/helm.go:196]: Can we also a test case for this?

Comment (gsquared94) [reply pkg/skaffold/deploy/helm.go:196]: are you saying that I might miss some `.lock` file or that I'll erroneously include it?
I'm currently removing the `.lock` files by name without depending on the path and AFAIK `Chart.lock` and `requirements.lock` are colocated with the respective `.yaml` files, which is always in `chartPath

Comment (gsquared94) [reply pkg/skaffold/deploy/helm.go:196]: discussed offline

**Pull request merged**